src/acl/external/kerberos_sid_group/ext_kerberos_sid_group_acl.pl.in

```diff
@@ -0,0 +1,252 @@
+my $user;
+my @groups;
+my $ans;
+
+# Disable output buffering
+$|=1;
+
+sub debug()
+{
+	my @lt = localtime;
+	print STDERR strftime("%Y/%m/%d %H:%M:%S", @lt)." | $name: @_\n" if $opt{d};
+}
+
+sub info()
+{
+	my @lt = localtime;
+	print STDERR strftime("%Y/%m/%d %H:%M:%S", @lt)." | $name: @_\n"; 
+}
+
+sub check()
+{
+	if ( grep( /^@_$/, @ADgroupSIDs) ) {
+		&debug("DEBUG: Found @_ in AD group SID");
+		return "OK";
+	} else {
+		&debug("DEBUG: Did not find @_ in AD group SID");
+		return "ERR";
+	}   
+}
+
+#
+# Command line options processing
+#
+sub init()
+{
+	use Getopt::Std;
+	my $errmsg;
+	my $opt_string = 'hdD:p:b:G:';
+	getopts( "$opt_string", \%opt ) or usage();
+	usage() if $opt{h};
+	usage() if not defined $opt{D};
+	usage() if not defined $opt{b};
+	usage() if not defined $opt{p};
+	usage() if not defined $opt{G};
+
+	$ENV{'KRB5CCNAME'} = $ccname;
+
+	@groups = split(/:/,$opt{G});
+	$errmsg=`kinit -k $opt{p} 2>&1`;
+	&info("ERROR: $errmsg") if $errmsg;
+	exit 99 if $errmsg;
+
+	$errmsg="";
+	foreach my $group (@groups) {
+		open(LDAP, "ldapsearch -LLL -Ygssapi -H ldap://$opt{D}:389 -s sub -b \"$opt{b}\" \"(CN=$group)\" objectsid 2>&1 |");
+		my $sid;
+		while (<LDAP>) {
+			chomp($_);
+			if ( $_ =~ /^object/ && defined $sid ) {
+				&info("ERROR: multiple SIDs returned for group $group");
+			} elsif ( $_ =~ /^object/ ) {
+				$sid=$_;
+				$sid=~s/^[^\s]+\s+//;
+			} else {
+				$errmsg=$errmsg.";".$_;
+			}
+		}
+		close(LDAP);
+		if ( ! defined $sid ) {
+			$errmsg=~s/^;//;
+			&info("ERROR: $errmsg");
+			&info("ERROR: no SID returned for group $group");
+		} else {
+			&info("INFO:ldapsearch result Group=$group, SID=$sid");
+			push @ADgroupSIDs, $sid;
+		}
+	}
+	&info("ERROR: Exit as no sid was found for any group") if ! @ADgroupSIDs;
+	exit 99 if ! @ADgroupSIDs;
+}
+
+#
+# Message about this program and how to use it
+#
+sub usage()
+{
+        print "Usage: kerberos_sid_group_acl -dh\n";
+        print "\t-d enable debugging\n";
+        print "\t-h print the help\n";
+        print "\t-p principal name (required)\n";
+        print "\t-D domain controller (required)\n";
+        print "\t-b base dn (required)\n";
+        print "\t-G list of AD group names (seperated by : ) (required)\n";
+        exit;
+}
+
+init();
+&debug("INFO: Debugging mode ON.");
+
+
+#
+# Main loop
+#
+while (<STDIN>) {
+        chop;
+        &debug("DEBUG: Got $_ from squid");
+        ($user, $groupSIDs) = split(/\s+/);
+        if ( defined $user && defined $groupSIDs ) {
+        	&debug("DEBUG: user=$user");
+        	&debug("DEBUG: groups=$groupSIDs");
+        	# test for each group squid send in it's request
+        	foreach my $group (split(/,/,$groupSIDs)) {
+                	$ans = &check($group);
+                	last if $ans eq "OK";
+        	}
+        	&debug("DEBUG: Sending $ans to squid");
+        	print "$ans\n";
+        } else {
+        	&debug("DEBUG: Sending ERR to squid");
+        	print "ERR\n";
+	}
+}
```

Comment:
our tools unfortunately need an empty line at the end of files. please add one here.